### PR TITLE
SLVS-1183 SLVS goldbars are not reset on solution/binding change

### PR DESCRIPTION
**Pull request by vnaskos-sonar**
[SLVS-1183](https://sonarsource.atlassian.net/browse/SLVS-1183)

Expectations:

1. On a project with shared binding which is still in Standalone mode, after binding the notification should close
2. On a project which is still in Standalone mode and an Open in IDE request came, after binding the notification to connect should close

[SLVS-1183]: https://sonarsource.atlassian.net/browse/SLVS-1183?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/SLCore.UnitTests/NodeJS/Notifications/UnsupportedNodeVersionNotificationServiceTests.cs [hunk 1]
```diff
@@ -21,6 +21,7 @@
 using SonarLint.VisualStudio.Core;
 using SonarLint.VisualStudio.Core.Notifications;
 using SonarLint.VisualStudio.SLCore.NodeJS.Notifications;
+using Language = SonarLint.VisualStudio.SLCore.Common.Models.Language;
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SLCore.UnitTests/NodeJS/Notifications/UnsupportedNodeVersionNotificationServiceTests.cs:24]: Was this needed?

File: src/Integration/MefServices/SharedBindingSuggestionService.cs [hunk 2]
```diff
@@ -88,6 +91,18 @@ private void OnActiveSolutionChanged(object sender, ActiveSolutionChangedEventAr
             {
                 Suggest();
             }
+            else
+            {
+                suggestSharedBindingGoldBar.Close();
+            }
+        }
+        
+        private void OnActiveSolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs e)
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration/MefServices/SharedBindingSuggestionService.cs:100]: Ideally this should be split into a different PR

File: src/Core/Notifications/INotificationService.cs [hunk 3]
```diff
@@ -45,28 +47,36 @@ public interface INotificationService : IDisposable
...
     private readonly IDisabledNotificationsStorage notificationsStorage;
+    private readonly IActiveSolutionTracker activeSolutionTracker;
     private readonly IThreadHandling threadHandling;
     private readonly ILogger logger;
 
     private readonly HashSet<string> oncePerSessionNotifications = [];
 
     private AttachedNotification activeNotification;
 
+    public bool CloseOnSolutionClose { get; set; } = true;
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Core/Notifications/INotificationService.cs:61]: Shouldn't this be a property of the notification instead?

File: src/ConnectedMode/Binding/Suggestion/SuggestSharedBindingGoldBar.cs [hunk 4]
```diff
@@ -53,6 +53,10 @@ public SuggestSharedBindingGoldBar(INotificationService notificationService,
             this.doNotShowAgainNotificationAction = doNotShowAgainNotificationAction;
             this.solutionInfoProvider = solutionInfoProvider;
             this.browserService = browserService;
+            
+            // Prevent INotificationService from handling the on solution change events
+            // as this is exceptionally overriden by the SharedBindingSuggestionService
+            this.notificationService.CloseOnSolutionClose = false;
```
Comment (georgii-borovinskikh-sonarsource) [inline src/ConnectedMode/Binding/Suggestion/SuggestSharedBindingGoldBar.cs:59]: Not a fan of this approach of setting the value directly on the service instead of the notification

File: src/Core/Notifications/INotificationService.cs (hunk 3 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/Core/Notifications/INotificationService.cs:61]: Also, could rename to CloseOnSolutionChange

**Review by georgii-borovinskikh-sonarsource — CHANGES_REQUESTED** [review]
Move CloseOnSolutionClose property to INotification

File: src/SLCore.UnitTests/NodeJS/Notifications/UnsupportedNodeVersionNotificationServiceTests.cs (hunk 1 above)
Comment (vnaskos-sonar) [reply src/SLCore.UnitTests/NodeJS/Notifications/UnsupportedNodeVersionNotificationServiceTests.cs]: It was automatically applied on commit because of the "Optimize imports" setting on Rider

File: src/Integration/MefServices/SharedBindingSuggestionService.cs (hunk 2 above)
Comment (vnaskos-sonar) [reply src/Integration/MefServices/SharedBindingSuggestionService.cs:100]: Created a separate PR for the close notification on solution close https://github.com/SonarSource/sonarlint-visualstudio/pull/5781

File: src/Core/Notifications/INotificationService.cs (hunk 3 above)
Comment (vnaskos-sonar) [reply src/Core/Notifications/INotificationService.cs]: Moved it to the INotification, that was true, it's much cleaner solution.

The name is fitting because we only care about solution close.

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]
